Add Cookie Effects

rmc opened the pull request:
Added effect that allows to read, write, and delete cookies

okwolf commented:
Thanks for all the effort you've taken to write, test, and document this. I'm ok with having FX for interacting with `document.cookie`. I'm less certain that I'd like to own and maintain the code for encoding and decoding cookies since those are essentially pure string transformation functions that don't really need any effectful API. Hopefully there are utilities already available for this.

I also feel like `localStorage` might be higher priority and slightly more modern FX to support. Are you proposing this because you have a legacy app that needs cookies to work?

rmc commented:
> Thanks for all the effort you've taken to write, test, and document this. I'm ok with having FX for interacting with `document.cookie`. I'm less certain that I'd like to own and maintain the code for encoding and decoding cookies since those are essentially pure string transformation functions that don't really need any effectful API. Hopefully there are utilities already available for this.

I can understand your reticence to owning code. I've implemented the cookie FX this way so that I don't have to worry about correctly decoding cookie values in the action that receives the cookie value. And because of symmetry, I also don't have to worry about correctly encoding the cookie name and value whenever a cookie is written.

> I also feel like `localStorage` might be higher priority and slightly more modern FX to support. 

I've implemented some Storage FX as well, but I haven't finished writing the tests. I'll send a separate PR for that when I've finished (almost there)

> Are you proposing this because you have a legacy app that needs cookies to work?

Let's not call it legacy, we'll just call it "Enterprise" 😜

okwolf commented:
Since you've included tests and documentation, I'm leaning toward bringing this in. I will be updating the examples in the docs for the other FX and would like you to follow suit. Speaking of docs - I'd like you to `npm run doc` as part of these changes so the auto-generated docs stay updated. If you rebase off the latest in `HAv2` this will happen automatically on install.

rmc commented:
> Since you've included tests and documentation, I'm leaning toward bringing this in. 

Brilliant, much appreciated.

> I will be updating the examples in the docs for the other FX and would like you to follow suit. 

I'll wait till I see the style of examples you want and try to replicate it.

> Speaking of docs - I'd like you to `npm run doc` as part of these changes so the auto-generated docs stay updated. If you rebase off the latest in `HAv2` this will happen automatically on install.

Done

okwolf commented:
New examples are available now. They might get some cleanup later, but should give you an idea what I'm thinking.
